spec/compiler/semantic/macro_overload_spec.cr

```diff
@@ -0,0 +1,29 @@
+require "../../spec_helper"
+
+describe "Semantic: macro overload" do
+  it "something between 2 macros (I need a name!!!)" do
+    assert_type(%(
+      macro foo(*, arg1)
+        {{ arg1 }}
```

Comment:
Perhaps use `1` and `1_i64` for these two so the spec can tell if the compiler uses the other overload.

Reply:
Rebased, and I used `1` and `"foo"`, as I find it easier to directly see the difference in type of these two literals!